algorithms/kernel/algorithm_base_impl.cpp

```diff
@@ -80,7 +80,7 @@ const data_management::DataCollectionPtr & algorithms::Argument::getStorage(cons
     return a._storage;
 }
 
-services::SharedPtr<Base> internal::ArgumentStorage::getExtension(Extension type)
+services::SharedPtr<Base> internal::ArgumentStorage::getExtension(Extension type) const
```

Comment:
Const member function, which returns pointer to non-const object looks like the bug. I think it should be 2 overloaded function, and const one must return pointer to const object.